compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -6088,6 +6095,13 @@ private boolean checkFillerValue(BUnionType type) {
                     hasFillerValue = true;
                 }
             } else {
+                if (member.tag == TypeTags.TUPLE) {
+                    for (BType tupleType: ((BTupleType) member).tupleTypes) {
+                        if (tupleType.tsymbol.getName().equals(type.tsymbol.getName())) {
+                            return false;
+                        }
+                    }
+                }
```

Comment:
This change done for https://github.com/ballerina-platform/ballerina-lang/issues/37049